README.md

```diff
@@ -1 +1,180 @@
-# Starcounter.Linq
\ No newline at end of file
+# Starcounter.Linq
+
+Starcounter.Linq is a LINQ to SQL Provider for Starcounter Database. It uses standard LINQ syntax for queries.
+
+It is available for downloading as [Starcounter.Linq](https://www.nuget.org/packages/Starcounter.Linq/) NuGet package.
+
+## Requirements
+
+Requires Starcounter version 2.3.1 and .NET Framework 4.5.
+
+## How to use
+
+Firstly, in your Starcounter app add a reference to `Starcounter.Linq.dll` through NuGet: `Install-Package Starcounter.Linq`.
+
+`DbLinq` static class defined in `Starcounter.Linq` namespace should be used as starting point when you want to build LINQ queries. It contains two important methods: `Objects<T>` and `CompileQuery<T>` (many overloaded ones) which are used for two different approaches. For example:
+
+```csharp
+// ad-hoc request
+var johns = DbLinq.Objects<Person>().Where(p => p.Name == "John");
+
+// request using compiled query
+var peopleByNameQuery = DbLinq.CompileQuery((string name) =>
+                        DbLinq.Objects<Person>().Where(p => p.Name == name));
+var jennifers = peopleByNameQuery("Jennifer");
+```
+
+## Ad-hoc requests
+
+Using this approach you build a LINQ expression and obtain data as usual when an application needs it.
+
+`DbLinq.Objects<T>()` returns a data context `Queryable<T>` which can be used for obtaining data.
+
+We recommend to use it only in cases when a request is expected to be executed infrequently. The reason is that it is very expensive to translate LINQ expression to SQL. Using this approach such translating will be done every calling.
+
+Example:
+
+```csharp
+// this method is used quite rare, so we can use ad-hoc requests here
+void Handle(Input.DeleteGroupTrigger action)
+{
+    Db.Transact(() =>
+    {
+        var surfaces = DbLinq.Objects<WebTemplate>().Where(x => x.WebTemplateGroup == this.Data);
+
+        foreach (WebTemplate surface in surfaces)
+        {
+            surface.WebTemplateGroup = null;
+        }
+        this.Data.Delete();
+    });
+}
+```
+
+## Compiled Query

```

Comment:
Is it intentional that Compiled Query always is written in title casing? I think "Query" should be lowercase